java-security-test/src/main/java/com/sap/cloud/security/test/JwtGenerator.java

```diff
@@ -255,6 +309,12 @@ public Token createToken() {
 		}
 	}
 
+	private void putScopesInJsonPayload() {
+		List<String> resultingScopes = Stream.concat(localScopes.stream(), scopes.stream())
+				.collect(Collectors.toList());
+		jsonPayload.put(TokenClaims.XSUAA.SCOPES, resultingScopes);
```

Comment:
can you replace
jsonPayload.put(TokenClaims.XSUAA.SCOPES, resultingScopes);
by
withClaimValues(TokenClaims.XSUAA.SCOPES, resultingScopes);
?